todoapp/app/src/test/java/com/example/android/architecture/blueprints/todoapp/tasks/TaskItemViewModelTest.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright 2016, The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.android.architecture.blueprints.todoapp.tasks;
+
+import android.content.Context;
+import android.content.res.Resources;
+
+import com.example.android.architecture.blueprints.todoapp.data.Task;
+import com.example.android.architecture.blueprints.todoapp.data.source.TasksDataSource;
+import com.example.android.architecture.blueprints.todoapp.data.source.TasksRepository;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import static junit.framework.Assert.assertEquals;
+import static junit.framework.Assert.assertFalse;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+/**
+ * Unit tests for the implementation of {@link TasksViewModel}
+ */
+public class TaskItemViewModelTest {
+
+    @Mock
+    private TasksRepository mTasksRepository;
+
+    @Mock
+    private Context mContext;
+
+    @Mock
+    private TaskItemNavigator mTaskItemNavigator;
+
+    @Captor
+    private ArgumentCaptor<TasksDataSource.GetTaskCallback> mLoadTasksCallbackCaptor;
+
+    private TaskItemViewModel mTaskItemViewModel;
+
+    private Task mTask;
+
+    @Before
+    public void setupTasksViewModel() {
+        // Mockito has a very convenient way to inject mocks by using the @Mock annotation. To
+        // inject the mocks in the test the initMocks method needs to be called.
+        MockitoAnnotations.initMocks(this);
+
+        // Get a reference to the class under test
+        mTaskItemViewModel = new TaskItemViewModel(mContext, mTasksRepository, mTaskItemNavigator);
+    }
+
+    @Before
+    public void setupContext() {
+        when(mContext.getResources()).thenReturn(mock(Resources.class));
+    }
+
+    @Test
+    public void clickOnTask_ShowsDetailUi() {
+        loadTaskIntoViewModel();
+
+        mLoadTasksCallbackCaptor.getValue().onTaskLoaded(mTask); // Trigger callback
```

Comment:
They're different comments. The When..Then explains the test, whereas that one is a detail of what that particular line does, as arg captors are tricky.